File: kamon-examples/kamon-autoweave-example/src/main/scala/kamon/autoweave/example/PingPong.scala
/*
 * =========================================================================================
 * Copyright © 2013-2015 the kamon project <http://kamon.io/>
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
 * except in compliance with the License. You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the
 * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 * =========================================================================================
 */

package kamon.autoweave.example

import akka.actor._
import kamon.Kamon

object PingPong extends App {
  Kamon.start()

  val system = ActorSystem()

  val pinger = system.actorOf(Props(new Actor {
    def receive: Actor.Receive = { case "pong" ⇒ sender ! "ping" }
  }), "pinger")

  val ponger = system.actorOf(Props(new Actor {
    def receive: Actor.Receive = { case "ping" ⇒ sender ! "pong" }
  }), "ponger")

  pinger.tell("pong", ponger)
}

//sbt run 
//or
//sbt assembly && java -jar target/scala-2.11/kamon-autoweave-example-assembly-0.1-SNAPSHOT.jar

File: kamon-examples/kamon-annotation-example/java/src/main/java/kamon/akka/pingpong/PingPong.java
package kamon.akka.pingpong;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;
import org.springframework.stereotype.Component;

import javax.annotation.PostConstruct;

@Component
public class PingPong {

    @PostConstruct
    public void initialize() {
        final ActorSystem system = ActorSystem.create("kamon-spring-boot-actor-system");

        final ActorRef pinger = system.actorOf(Props.create(Pinger.class), "pinger");
        final ActorRef ponger = system.actorOf(Props.create(Ponger.class), "ponger");

        pinger.tell(new Ponger.PongMessage(), ponger);
    }
}
